cogito/putter.go

```diff
@@ -83,14 +98,37 @@ func (putter *ProdPutter) ProcessInputDir() error {
 
 	params := putter.Request.Params
 	source := putter.Request.Source
+
+	sinks := sets.From(source.Sinks...)
+	if len(params.Sinks) > 0 {
```

Comment:
How can it be that `sinks.Size() == 0`? Either `params.Sinks` or `source.Sinks` will return defaults no?

Reply:
Not exact, this is the pipeline configuration, the default is returned in case sinks are not configured. Here  by default we set sinks from `source` setting, but if specified in `put` parameter, then we override.